### DISCUSSION TITLE
How to manually invoke failure of a workflow/task?

### DISCUSSION THREAD

**Discussion by mardonner** [StackStorm/st2#6284] [Q&A] [answered]
Hey,
coming from python I'm having a hard time wrapping my head around orquesta workflow definitions, loops and so on.

Example:
I construct and publish a variable in my first task. Then, in the second task, I want to compare the value with an expected value.
How do I make the second task fail, if the value is not equal to the expected value?
Basically, if condition true, continue with next task, else raise error and stop.

**Comment by nzlosh** [comment, marked as answer]
You can use the `when` clause to test the variables and call `do: fail`.  https://docs.stackstorm.com/orquesta/languages/orquesta.html#task-transition-model

**Reply by mardonner** [reply]
ty, somehow missed that in the docs
